EIPS/eip-6682.md

```diff
@@ -0,0 +1,161 @@
+
+In most cases, the handling of fee payments will be desired to be paid in a seperate currency to the loaned NFTs because NFTs themselves cannot always be fractionalized. Consider the following example where the flashloan provider charges a 0.1 ETH fee on each NFT that is flashloaned; The interface must provide methods that allow the borrower to determine the fee rate on each NFT and also the currency that the fee should be paid in.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Contract Interface
+
+```solidity
+pragma solidity ^0.8.19;
+
+interface IERC6682 {
+    /// @dev The address of the token used to pay flash loan fees.
+    function flashFeeToken() external view returns (address);
+
+    /// @dev Whether or not the NFT is available for a flash loan.
+    /// @param token The address of the NFT contract.
+    /// @param tokenId The ID of the NFT.
+    function availableForFlashLoan(address token, uint256 tokenId) external view returns (bool);
+}
+```
+
+The `flashFeeToken` function MUST return the address of the token used to pay flash loan fees.
+
+If the token used to pay the flash loan fees is ETH then `flashFeeToken` MUST return `address(0)`.
+
+The `availableForFlashLoan` function MUST return whether or not the `tokenId` of `token` is available for a flashloan. If the `tokenId` is not currently available for a flashloan `availableForFlashLoan` MUST return `false` instead of reverting.
+
+## Rationale
+
+The above modifications are the simplest possible additions to the existing flashloan standard to support NFTs.
+
+We choose to extend as much of the existing flashloan standard ([ERC-3156](./eip-3156.md)) as possible instead of creating a wholly new standard because the flashloan standard is already widely adopted and the changes required to support NFTs are minimal.
+
+## Backwards Compatibility
+
+This EIP is fully backwards compatible with [ERC-3156](./eip-3156.md) with the exception of the `maxFlashLoan` method. This method does not make sense within the context of NFTs because NFTs are not fungible. However it is part of the existing flashloan standard and so it is not possible to remove it without breaking backwards compatibility. It is RECOMMENDED that any contract implementing this EIP without the intention of supporting ERC-20 flashloans should always return `0` from `maxFlashLoan`. For example:
+
+```solidity
+function maxFlashLoan(address token) public pure override returns (uint256) {
+    // if a contract also supports flash loans for ERC20 tokens then it can
+    // return some value here instead of 0
+    return 0;
+}
+```
+
+## Reference Implementation
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code included inline in the document must be licensed under `CC0-1.0`. You can remove the comment, put the reference implementation in `../assets/eip-6682/`, or:

```suggestion
// SPDX-License-Identifier: CC0-1.0
```

Reply:
updated